routing/router.go

```diff
@@ -6,21 +6,26 @@ import (
 	"github.com/gorilla/mux"
 
 	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/config/security"
 )
 
 // NewRouter creates the router with all API routes and the static files handler
-func NewRouter(conf *config.Config) *mux.Router {
+func NewRouter(conf *config.Config, credentials security.Credentials) *mux.Router {
 
 	router := mux.NewRouter().StrictSlash(true)
 
 	// Build our API server routes and install them.
 	routes := NewRoutes()
 	for _, route := range routes.Routes {
+		var handlerFunction http.Handler
+		if handlerFunction = route.HandlerFunc; route.Authenticated {
+			handlerFunction = config.Authentication(handlerFunction, credentials)
```

Comment:
Side comment, not blocker as well.
Security and authentication will change, so, from the perspective of "today" it is unlikely we will have different credentials per router. So perhaps config.Authentication can read directly the credentials and no need to propragate them from the router.

In the future, when security is refactored, it can be updated, but I guess that credential propagation is not needed at the moment, then it could same some extra modifications in the code.

Reply:
I agree...I tried to do it like you proposed but there is a cycle import, I am gonna try to avoid this